tinygrad/tensor.py

```diff
@@ -642,7 +642,7 @@ def mul(self, x:Union[Tensor, float], reverse=False) -> Tensor:
     if x.__class__ is not Tensor and x == 0.0: return mlops.Zero.apply(self)
     if x.__class__ is not Tensor and x == -1.0: return -self
     return mlops.Mul.apply(*self._broadcasted(x, reverse)) if x.__class__ is Tensor or x != 1.0 else self
-  def div(self, x:Union[Tensor, float], reverse=False) -> Tensor: return mlops.Div.apply(*self._broadcasted(x, reverse)) if x.__class__ is Tensor or reverse or not x or not dtypes.is_float(self.dtype) else self.mul(1/x)
```

Comment:
This is not for correctness, this is for speed.

Reply:
Metal and LLVM do this at compile time (unless disabled) -  so you get the optimization anyways.  

If you do the reciprocal optimization here as well as in the compiler then changing the compiler flags has no effect.

Reply:
This is before mlops.

Reply:
I reverted it and the ONNX tests pass. The example in #1791 fails though (`(Tensor([3]) + 0.5) / 3.5`) - that only passes with this change.

I am now very confused as to why fast-math is causing the ONNX tests to fail.  Clearly not reciprocal division.  Maybe this:

![Screenshot 2023-10-24 at 2 40 38 PM](https://github.com/tinygrad/tinygrad/assets/4053345/5b60230b-bc62-4183-b42d-14d170415123)